Common/Tools/handleParamTPCResponse.cxx

```diff
@@ -0,0 +1,207 @@
+
+#include "CCDB/CcdbApi.h"
+#include <boost/program_options.hpp>
+#include <FairLogger.h>
+#include <array>
+#include "TFile.h"
+#include "Common/Core/PID/PIDResponse.h"
+#include "Common/Core/PID/TPCPIDResponse.h"
+using namespace o2::pid::tpc;
+namespace bpo = boost::program_options;
+
+bool initOptionsAndParse(bpo::options_description& options, int argc, char* argv[], bpo::variables_map& vm)
+{
+  options.add_options()(
+    "url,u", bpo::value<std::string>()->default_value("http://alice-ccdb.cern.ch"), "URL of the CCDB database")(
+    "ccdb-path,c", bpo::value<std::string>()->default_value("Analysis/PID/TPC/Response"), "CCDB path for object")(
+    "start,s", bpo::value<long>()->default_value(0), "Start timestamp for calibration validity")(
+    "stop,S", bpo::value<long>()->default_value(4108971600000), "End timestamp for calibration validity")(
+    "delete-previous", bpo::value<int>()->default_value(0), "delete previous entry from CCDB (1 = true)")(
+    "save-to-file,file,f,o", bpo::value<std::string>()->default_value(""), "Option to save parametrization to file instead of uploading to ccdb")(
+    "read-from-file,i", bpo::value<std::string>()->default_value(""), "Option to get parametrization from a file")(
+    "objname,n", bpo::value<std::string>()->default_value("Response"), "Object name to be stored in file")(
+    "bb0", bpo::value<float>()->default_value(0.0320981f), "Bethe-Bloch parameter 0")(
+    "bb1", bpo::value<float>()->default_value(19.9768f), "Bethe-Bloch parameter 1")(
+    "bb2", bpo::value<float>()->default_value(2.52666e-16f), "Bethe-Bloch parameter 2")(
+    "bb3", bpo::value<float>()->default_value(2.72123f), "Bethe-Bloch parameter 3")(
+    "bb4", bpo::value<float>()->default_value(6.08092f), "Bethe-Bloch parameter 4")(
+    "sig0", bpo::value<float>()->default_value(0.07f), "Sigma parameter 0")(
+    "sig1", bpo::value<float>()->default_value(0.f), "Sigma parameter 1")(
+    "paramMIP", bpo::value<float>()->default_value(50.f), "MIP parameter value")(
+    "paramChargeFactor", bpo::value<float>()->default_value(2.3f), "Charge factor value")(
+    "mode", bpo::value<string>()->default_value(""), "Running mode ('read' from file, 'write' to file, 'pull' from CCDB, 'push' to CCDB)")(
+    "help,h", "Print this help.");
+  try {
+    bpo::store(parse_command_line(argc, argv, options), vm);
+
+    // help
+    if (vm.count("help")) {
+      LOG(info) << options;
+      return false;
+    }
+    bpo::notify(vm);
+  } catch (const bpo::error& e) {
+    LOG(error) << e.what() << "\n";
+    LOG(error) << "Error parsing command line arguments; Available options:";
+    LOG(error) << options;
+    return false;
+  }
+  return true;
+
+} //initOptionsAndParse
+
+int main(int argc, char* argv[])
+{
+  bpo::options_description options("Allowed options");
+  bpo::variables_map vm;
+  if (!initOptionsAndParse(options, argc, argv, vm)) {
+    return 1;
+  }
+
+  Response* tpc = nullptr;
```

Comment:
there is no delete on `tpc`. This is a memory leak...
So either make this a `std::unique_ptr<Response> tpc` or call delete properly. I would prefer unique_ptr with the changes below.